pep-0517.txt

```diff
@@ -503,6 +540,48 @@ package *authors*, while still allowing *end-users* to open up the
 hood and apply duct tape when necessary.
 
 
+In-Tree Build Backends
+======================
+
+In certain circumstances, projects may wish to include the source code for the
+build backend directly in the source tree, rather than referencing the backend
+via the ``requires`` key. Two specific situations where this would be expected
+are:
+
+- Backends themselves, which want to use their own features for building
+  themselves ("self-hosting backends")
+- Project-specific backends, typically consisting of a custom wrapper around a
+  standard backend, where the wrapper is too project-specific to be worth
+  distributing independently ("in-tree backends")
+
+Projects can specify that their backend code is hosted in-tree by including the
+``python-paths`` key in ``pyprject.toml``. This key contains a list of
+directories, which the frontend will make available on ``sys.path`` when
+loading the backend, and running the backend hooks. The directories in
+``python-paths`` will be added to the *start* of ``sys.path``.
+
+There are a number of restrictions on the content of the ``python-paths`` key:
+
+- Directories in ``python-paths`` are interpreted as relative to the project
+  root, and MUST resolve to the project root or a subdirectory of it.
```

Comment:
"Resolve" is slightly ambiguous, It might make sense to call out that this includes both symlinks and relative paths.

Reply:
Agreed. I wondered about this, but wasn't sure what would be good wording. I'll have a go at something more explicit.

Reply:
The paragraph below, saying that frontends SHOULD enforce this check, makes this clearer. I've added a note there saying that symlinks should be resolved, and simplified the statement here to just say "MUST refer to a location within the source tree" ("refer" seems clearer than "resolve").

If the result still isn't sufficient, feel free to suggest something better. I don't really know how complicated and messy things can get - there are probably ways to get around this check if someone really cared enough, but I don't think that matters if the intent is clear.